supervisor/api/utils.py

```diff
@@ -116,6 +118,28 @@ async def wrap_api(api, *args, **kwargs):
     return wrap_method
 
 
+def api_process_custom(content_type):
+    """Ensure errors are handled and returned with specified content_type."""
+
+    def decorator(method):
+        async def wrapper(api, *args, **kwargs):
+            status = 200
+            try:
+                response = await method(api, *args, **kwargs)
+            except HassioError as err:
+                response = str(err)
+                status = 400
```

Comment:
```suggestion
            except APIError as err:
                response = str(err)
                status = err.status
            except HassioError as err:
                response = str(err)
                status = 400
```
This way if a method raises an `APIForbidden` (or if we make other HTTP Status codes into exceptions in the future) it will use the correct status code.